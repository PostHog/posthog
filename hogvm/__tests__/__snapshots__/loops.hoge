["_h", 32, "-- test while loop --", 2, "print", 1, 35, 33, 0, 33, 3, 36, 0, 15, 40, 15, 33, 1, 36, 0, 6, 37, 0, 36, 0,
2, "print", 1, 35, 39, -22, 36, 0, 2, "print", 1, 35, 35, 32, "-- test for loop --", 2, "print", 1, 35, 33, 0, 33, 3,
36, 0, 15, 40, 15, 36, 0, 2, "print", 1, 35, 33, 1, 36, 0, 6, 37, 0, 39, -22, 35, 32, "i", 1, 1, 2, "print", 1, 35, 32,
"-- test emptier for loop --", 2, "print", 1, 35, 33, 0, 33, 3, 36, 0, 15, 40, 15, 32, "woo", 2, "print", 1, 35, 33, 1,
36, 0, 6, 37, 0, 39, -22, 32, "hoo", 2, "print", 1, 35, 35]
