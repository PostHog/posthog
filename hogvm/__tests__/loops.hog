print('-- test while loop --')
{
    let i := 0
    while (i < 3) {
        i := i + 1
        print(i)
    }
    print(i)
}
