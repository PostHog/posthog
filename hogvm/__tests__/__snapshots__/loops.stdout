-- test while loop --
1
2
3
3
-- test for loop --
0
1
2
null
-- test emptier for loop --
woo
woo
woo
hoo
-- for in loop with arrays --
1
2
3
-- for in loop with arrays and keys --
0 1
1 2
2 3
-- for in loop with tuples --
1
2
3
-- for in loop with tuples and keys --
0 1
1 2
2 3
-- for in loop with dicts --
v1
v2
v3
-- for in loop with dicts and keys --
first v1
second v2
third v3
